odl/solvers/functional/default_functionals.py

```diff
@@ -118,6 +121,114 @@ def convex_conj(self):
+        \\int_{\Omega} \\left(\\sum_{j = 1}^m x_{j}^p\\right)^{1/p} dx.
+    """
+
+    def __init__(self, vfspace, exponent=None, weight=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        vfspace : `ProductSpace`
+            Space of vector fields on which the operator acts.
+            It has to be a product space of identical spaces, i.e. a
+            power space.
+        exponent : non-zero float, optional
+            Exponent of the norm in each point. Values between
+            0 and 1 are currently not supported due to numerical
+            instability.
+            Default: ``vfspace.exponent``, usually 2.
+        weight : `array-like` or float, optional
+            Weighting array or constant for the pointwise norm. If an array is
+            given, its length must be equal to ``domain.size``, and
+            all entries must be positive. A provided constant must be
+            positive.
+            By default, the weights are is taken from
+            ``domain.weighting``. Note that this excludes unusual
+            weightings with custom inner product, norm or dist.
+        """
+        if not isinstance(vfspace, ProductSpace):
+            raise TypeError('`space` must be a `ProductSpace`')
+        if vfspace.is_power_space:
+            raise TypeError('`space.is_power_space` must be `True`')
+        self.pointwise_norm = PointwiseNorm(vfspace, exponent, weight)
+        super().__init__(space=vfspace, linear=False, grad_lipschitz=np.nan)
+
+    def _call(self, x):
+        """Return the L1-norm of ``x``."""
+        # TODO: update when integration operator is in place: issue #440
+        pointwise_norm = self.pointwise_norm(x)
+        return pointwise_norm.inner(self.domain.one())
+
+    # TODO: remove inner class when ufunc operators are in place: issue #567
+    @property
+    def gradient(self):
+        """Gradient operator of the functional.
+
+        The functional is not differentiable in ``x=0``. However, when
+        evaluating the gradient operator in this point it will return 0.
+        """
+        functional = self
+
+        class GroupL1Gradient(Operator):
+
+            """The gradient operator of the `GroupL1Norm` functional."""
+
+            def __init__(self):
+                """Initialize a new instance."""
+                super().__init__(functional.domain, functional.domain,
+                                 linear=False)
+
+            def _call(self, x):
+                """Apply the gradient operator to the given point."""
+                pointwise_norm = functional.pointwise_norm(x)
+                return x / pointwise_norm
+
+        return GroupL1Gradient()
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the functional.
+
+        See Also
+        --------
+        proximal_l1 : proximal factory for the L1-norm.
+        """
+        return proximal_l1(space=self.domain, isotropic=True)
```

Comment:
Not correct either. You rather get in each point the proximal of the p-norm, which is the identity minus the projection onto the p'-ball. That projection is not computable in closed form except for p = 1, 2 or infty. We probably don't need arbitrary p now, but at least the well-known cases should be correct now.